samza-kv/src/main/scala/org/apache/samza/storage/kv/SerializedKeyValueStore.scala

```diff
@@ -62,7 +62,10 @@ class SerializedKeyValueStore[K, V](
     val keyBytes = toBytesOrNull(key, keySerde)
     val valBytes = toBytesOrNull(value, msgSerde)
     store.put(keyBytes, valBytes)
+    val keySizeBytes = if (keyBytes == null) 0 else keyBytes.length
     val valSizeBytes = if (valBytes == null) 0 else valBytes.length
+    metrics.recordKeySizeBytes.update(keySizeBytes)
+    metrics.recordValueSizeBytes.update(valSizeBytes)
     updatePutMetrics(1, valSizeBytes)
```

Comment:
Is there a way we can roll together recording maximum record size into the histogram tracking? When I implemented the metric for recording max I had considered modifying histogram to record maximum in addition to percentiles, but decided against it since the memory footprint of the histogram was unjustifiable at the time. Now that we'd like those metrics anyway, maybe it is justifiable now.

Reply:
I thought about that before. But maybe there are some people using the maximum record size metric and deleting it may caused some trouble. So I choose to keep this metric and do not combine it with the histogram.

Reply:
I see. We wouldn't be able to preserve the name if we emitted a maximum metric from the histogram?

Reply:
I am not very sure about that because I am not very familiar with the code yet. I will keep that in mind and try to optimize it in the future if possible.